src/kakarot/instructions/environmental_information.cairo

```diff
@@ -91,10 +92,7 @@ namespace EnvironmentalInformation {
         let (stack: model.Stack*, address: Uint256) = Stack.pop(ctx.stack);
 
         // Get the starknet account address from the evm account address
-        let (registry_address_) = registry_address.read();
-        let (starknet_contract_address) = IRegistry.get_starknet_contract_address(
-            contract_address=registry_address_, evm_contract_address=address.low
-        );
+        let (starknet_contract_address) = Deployer.compute_starknet_address(address.low);
```

Comment:
if was most probably buggy before because an `address` is 20 bytes (a `uint160`). You need to use `Helpers.uint256_to_felt`

Reply:
we need to use a `IDeployer` together with a `deployer_address` because the deployer is not part of kakarot contract.
We could however update `kakarot.cairo` to deploy both EOA and CA though